pkg/action/failover.go

```diff
@@ -0,0 +1,384 @@
+		return
+	}
+	// check accessibility of the source cluster, get the migrationSchedule in the source cluster to check for any ongoing migrations and to suspend the migrationSchedule
+	if !ac.isClusterAccessible(action, sourceConfig) {
+		msg := "Unable to access the remote cluster. Directly moving to the Failover ScaleUpDestinationStage"
+		log.ActionLog(action).Warnf(msg)
+		// logging the Initial stage as successful and directly moving next stage to ScaleUpDestination
+		ac.recorder.Event(action,
+			v1.EventTypeWarning,
+			string(storkv1.ActionStatusSuccessful),
+			msg)
+		action.Status.Stage = storkv1.ActionStageScaleUpDestination
+		action.Status.Status = storkv1.ActionStatusInitial
+		action.Status.FinishTimestamp = metav1.Now()
+		ac.updateAction(action)
+		return
+	}
+
+	remoteOps, err := storkops.NewForConfig(sourceConfig)
+	if err != nil {
+		msg := fmt.Sprintf("Failed to create the stork client to access cluster with the config Host : %v", sourceConfig.Host)
+		logEvents := ac.printFunc(action, string(storkv1.ActionStatusFailed))
+		logEvents(msg, "err")
+		action.Status.Status = storkv1.ActionStatusFailed
+		action.Status.Reason = msg
+		ac.updateAction(action)
+		return
+	}
+	srcMigrSched, err := remoteOps.GetMigrationSchedule(action.Spec.ActionParameter.FailoverParameter.MigrationScheduleReference, action.Namespace)
+	if err != nil {
+		msg := fmt.Sprintf("Unable to find MigrationSchedule %s/%s for failover. Skipping MigrationSchedule suspension : %v", migrationSchedule.Namespace, migrationSchedule.Name, err)
+		log.ActionLog(action).Warnf(msg)
+		ac.recorder.Event(action,
+			v1.EventTypeWarning,
+			string(storkv1.ActionStatusSuccessful),
+			msg)
+		action.Status.Status = storkv1.ActionStatusSuccessful
+		action.Status.Reason = ""
+		ac.updateAction(action)
+		return
+	}
+
+	_, latestMigration := getLatestMigrationPolicyAndStatus(*srcMigrSched)
+	// Suspend the migration schedule in the source cluster
+	suspend := true
+	if !*srcMigrSched.Spec.Suspend {
+		srcMigrSched.Spec.Suspend = &suspend
+		msg := fmt.Sprintf("Suspending migration schedule %s/%s before doing the failover operation", srcMigrSched.Namespace, srcMigrSched.Name)
+		log.ActionLog(action).Infof(msg)
+		_, err := remoteOps.UpdateMigrationSchedule(srcMigrSched)
+		if err != nil {
+			log.ActionLog(action).Errorf("Error suspending MigrationSchedule %s/%s: %v", srcMigrSched.Namespace, srcMigrSched.Name, err)
+			return
+		}
+	}
+	if latestMigration != nil {
+		if !isMigrationComplete(latestMigration.Status) {
+			// a migration is still in progress. failover needs to wait
+			msg := fmt.Sprintf("Waiting for the completion of migration %s/%s", srcMigrSched.Namespace, latestMigration.Name)
+			log.ActionLog(action).Infof(msg)
+			ac.recorder.Event(action,
+				v1.EventTypeWarning,
+				string(storkv1.ActionStatusScheduled),
+				msg)
+			ac.updateAction(action)
+			return
+		} else if latestMigration.Status == storkv1.MigrationStatusFailed {
```

Comment:
Is it possible that only one migration happend and it failed. And we will still trigger a failover instead of showing an error?

Reply:
Yes it is indeed a possibility as per current code. 
Should we then block the failover instead in last migration failed scenarios ?? 
I guess this will be a problem only in controlled failover cases where the customer can still try and debug the migration failure because in disaster situation if we assume that source is inaccessible anyways these checks are skipped and scale up is done on destination.